api/python/provisioner/commands/upgrade/set_swupgrade_repo.py

```diff
@@ -652,18 +652,27 @@ def _pre_repo_validation(self, params: inputs.SWUpgradeRepo,
             Raise this exception if candidate repository validation fails
 
         """
-        if not params.is_remote():
-            if params.sig_file:
-                self._check_iso_authenticity(params)
-            elif params.hash:
+        if not dry_run and not params.is_remote():
+            if params.sig_file is None:
+                logger.info("Signature file is not specified")
+                suffix = params.source.suffix + ".sig"
+                params.sig_file = Path(str(params.source)).with_suffix(suffix)
+
+                if not params.sig_file.exists():
+                    raise SWUpdateRepoSourceError(
+                        str(params.source),
+                        f"Signature file '{params.sig_file}' is not found"
+                    )
+
+            self._check_iso_authenticity(params)
+
+            if params.hash:
```

Comment:
please fail here if not a dry run otherwise someone will be able to bypass the authenticity check

Reply:
Done.